PWGHF/Tasks/HFCorrelatorD0D0bar.cxx

```diff
@@ -436,14 +436,22 @@ struct HfCorrelatorD0D0bar {
             if (std::abs(particle1.eta()) < etaCut && std::abs(particle2.eta()) < etaCut && particle1.pt() > ptCut && particle2.pt() > ptCut) { // fill with D and Dbar acceptance checks
               registry.fill(HIST("hDDbarVsEtaCut"), etaCut - epsilon, ptCut + epsilon);
             }
-            if (rightDecayChannels) { // fill with D and Dbar daughter particls acceptance checks
-              double etaCandidate1Daughter1 = particle1.daughter0_as<MCParticlesPlus>().eta();
-              double etaCandidate1Daughter2 = particle1.daughter1_as<MCParticlesPlus>().eta();
-              double etaCandidate2Daughter1 = particle2.daughter0_as<MCParticlesPlus>().eta();
-              double etaCandidate2Daughter2 = particle2.daughter1_as<MCParticlesPlus>().eta();
-              if (std::abs(etaCandidate1Daughter1) < etaCut && std::abs(etaCandidate1Daughter2) < etaCut &&
-                  std::abs(etaCandidate2Daughter1) < etaCut && std::abs(etaCandidate2Daughter2) < etaCut &&
-                  particle1.pt() > ptCut && particle2.pt() > ptCut) {
+            if (rightDecayChannels) { //fill with D and Dbar daughter particls acceptance checks
+              bool candidate1DauInAcc = true;
+              bool candidate2DauInAcc = true;
+              for (auto& dau : particle1.daughters_as<MCParticlesPlus>()) {
+                if (std::abs(dau.eta()) > etaCut) {
+                  candidate1DauInAcc = false;
+                  break;
+                }
+              }
+              for (auto& dau : particle2.daughters_as<MCParticlesPlus>()) {
+                if (std::abs(dau.eta()) > etaCut) {
+                  candidate2DauInAcc = false;
+                  break;
+                }
+              }
+              if (candidate1DauInAcc && candidate2DauInAcc && particle1.pt() > ptCut && particle2.pt() > ptCut) {
```

Comment:
@fcolamar Can you please check these changes in your tasks and confirm that you are fine with them?